### PR TITLE
Support OIDC groups

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit enable project admin to add group as project member when
Harbor is configured against OIDC as AuthN backend.

It populates the information of groups from ID Token based on the claim
that is set in OIDC settings.

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: src/core/controllers/oidc.go [hunk 1]
```diff
@@ -114,6 +116,7 @@ func (oc *OIDCController) Callback() {
 		oc.SendInternalServerError(err)
 		return
 	}
+	d.GroupIDs, err = group.GetGroupIDByGroupName(oidc.GroupsFromToken(idToken), common.OIDCGroupType)
```
Comment (stonezdj) [inline src/core/controllers/oidc.go]: There is an err, it should be handled, or it will be overwrite by next line

Comment (reasonerjt) [reply src/core/controllers/oidc.go]: I'll just log this error but set empty list in the model.

File: src/core/controllers/oidc.go [hunk 2]
```diff
@@ -175,6 +182,7 @@ func (oc *OIDCController) Onboard() {
 		oc.SendBadRequestError(errors.New("Failed to get OIDC user info from session"))
 		return
 	}
+	defer oc.DelSession(userInfoKey)
```
Comment (wy65701436) [inline src/core/controllers/oidc.go]: in the previous implementation, line 223 will not delete the session.  But in this PR, it will.

Comment (reasonerjt) [reply src/core/controllers/oidc.go]: Will fix

**Pull request merged**